fs/copy_linux.go

```diff
@@ -108,3 +108,11 @@ func copyXAttrs(dst, src string) error {
 
 	return nil
 }
+
+func copyDevice(dst string, fi os.FileInfo) error {
+	st, ok := fi.Sys().(*syscall.Stat_t)
+	if !ok {
+		return errors.New("unsupported stat type")
+	}
+	return syscall.Mknod(dst, uint32(fi.Mode().Perm()), int(st.Rdev))
```

Comment:
I think it needs the full mode, the mode should also encodes the type of device to be created